app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/coverage/PerilBottomSheet.kt

```diff
@@ -1,69 +1,165 @@
+        val defaultSystemUiVisibility = dialog?.window?.decorView?.systemUiVisibility
+        val peril = requireArguments().getParcelable<Peril>(PERIL)
+
         binding.apply {
-            val titleText = requireArguments().getString(TITLE)
-            val bodyText = requireArguments().getString(BODY)
-            val iconUrl = requireArguments().getString(ICON_URL)
+            close.alpha = 0f
+            (dialog as? BottomSheetDialog)?.behavior?.let { behaviour ->
+                val peekHeight = 380.dp
+                behaviour.setPeekHeight(peekHeight, true)
+                chevronContainer.measure(
+                    FrameLayout.LayoutParams.MATCH_PARENT,
+                    FrameLayout.LayoutParams.WRAP_CONTENT
+                )
+                val chevronContainerHeight = chevronContainer.measuredHeight
+                val startTranslation = (peekHeight - chevronContainerHeight).toFloat()
+                chevronContainer.translationY = startTranslation
+                behaviour.addBottomSheetCallback(
+                    object : BottomSheetCallback() {
 
-            if (titleText == null || bodyText == null || iconUrl == null) {
-                e { "Programmer error: Missing either TITLE, BODY or ICON_URL in ${this.javaClass.name}" }
-                return
+                        override fun onStateChanged(bottomSheet: View, newState: Int) {
+                            when (newState) {
+                                BottomSheetBehavior.STATE_EXPANDED -> {
+                                    dialog?.window?.statusBarColor =
+                                        requireContext().colorAttr(R.attr.colorSurface)
+                                    if (!requireContext().isDarkThemeActive) {
+                                        dialog?.window?.decorView?.let {
+                                            it.systemUiVisibility =
+                                                it.systemUiVisibility or View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
+                                        }
+                                    }
+
+                                    close.setHapticClickListener {
+                                        this@PerilBottomSheet.dismiss()
+                                    }
+                                }
+                                STATE_DRAGGING -> {
+                                    defaultStatusBarColor?.let {
+                                        dialog?.window?.statusBarColor = it
+                                    }
+                                    defaultSystemUiVisibility?.let {
+                                        dialog?.window?.decorView?.systemUiVisibility = it
+                                    }
+                                }
+                                BottomSheetBehavior.STATE_COLLAPSED -> {
+                                    close.setOnClickListener(null)
+                                }
```

Comment:
```suggestion
```

Reply:
Aah was this the part you were talking about before? I thought it was the line that removed the dim 😄 This is needed or else there is an invisible X that can be pressed if the sheet was once expanded and then draged back to collapsed state

Reply:
How come? 🤔 Should we not control this using visibility?

Reply:
Additionally, we should be able to set it once, and not each time we expand/collapse the sheet, right? 🤔

Reply:
Good point visibility gone is much better here